torch_frame/data/multi_nested_tensor.py

```diff
@@ -65,6 +65,13 @@ def validate(self):
         assert self.offset[-1] == len(self.values)
         assert len(self.offset) == self.num_rows * self.num_cols + 1
 
+    def __eq__(self, td: object) -> bool:
```

Comment:
FYI: This is a bit different from how PyTorch tensors deal with equality. If we want to keep consistency, `__eq__` would need to return a mask, and equality would need to be checked via `tensor.equal()`/`torch.eq` and `tensor.allclose()`.

Reply:
I feel like returning a mask would be inefficient here. Because then we need to convert the two `MultiTensor`s to their dense format. The current implementation is faster. Or do you mean return a tuple of mask, one for the `values` and one for the `offset`?

Reply:
I think we should try to keep consistent with PyTorch functions.

Reply:
I am just saying you should rename this function to `eq` and `equal` :)